ucp/_libs/ucx_api.pyx

```diff
@@ -786,6 +791,17 @@ def tag_send_nb(
         cb_kwargs = {}
     if name is None:
         name = "tag_send_nb"
+    if buffer.cuda and not ep.worker._context.cuda_support:
```

Comment:
I think UCX will throw an error first if you don't define a CUDA transport while trying to pass CUDA data.  @pentschev is that right ?

Reply:
Well we had a test for this before. It was just in Python. Now we have moved it to C. I suppose we could drop the check, but I’m guessing it was there for a reason. So kept it in. Plus this check should be pretty fast

Reply:
For completeness: yes, UCX will fail, but we can't recover from C errors at the moment, so keeping them in UCX-Py is useful, regardless in what layer we do so.